backend/corpora/common/utils/db_utils.py

```diff
@@ -29,3 +31,14 @@ def query(self, table_args: typing.List[Base], filter_args: typing.List[bool] =
             if filter_args
             else self.session.query(*table_args).all()
         )
+
+    def commit(self):
+        """
+        Commit changes to the database and roll back if error.
+        :return:
+        """
+        try:
+            self.session.commit()
+        except SQLAlchemyError:
+            self.session.rollback()
+            raise
```

Comment:
I'll add a log and raise a corpora exception. Though I think this will be better handled once we have some logging standards in place. I'm a fan of using JSON formatted logs myself. They are easier to search and parse.